tests/ttnn/unit_tests/operations/test_moreh_nll_loss_unreduced.py

```diff
@@ -18,48 +18,56 @@
 )
 
 
-def get_torch_tensors(shape):
+def get_torch_tensors(shape, torch_dtype):
     C = shape[1]
     target_shape = shape[:1] + shape[2:]
 
-    cpu_dtype = torch.float32
     cpu_index_dtype = torch.long
 
-    torch_input = torch.rand(shape, dtype=cpu_dtype).requires_grad_()
+    torch_input = torch.rand(shape, dtype=torch_dtype).requires_grad_()
     torch_target = torch.randint(0, C, target_shape, dtype=cpu_index_dtype)
-    torch_weight = torch.rand(C, dtype=cpu_dtype)
-    torch_output = torch.empty(target_shape, dtype=cpu_dtype)
+    torch_weight = torch.rand(C, dtype=torch_dtype)
+    torch_output = torch.empty(target_shape, dtype=torch_dtype)
 
     return torch_input, torch_target, torch_weight, torch_output
 
 
-def get_tt_tensors(torch_input, torch_target, torch_weight, torch_output, device):
+def get_tt_tensors(torch_input, torch_target, torch_weight, torch_output, device, ttnn_dtype):
     npu_index_dtype = ttnn.int32
 
-    tt_input = to_ttnn(torch_input, device=device)
+    tt_input = to_ttnn(torch_input, device=device, dtype=ttnn_dtype)
     tt_target = to_ttnn(torch_target, device=device, dtype=npu_index_dtype)
-    tt_weight = to_ttnn(torch_weight, device=device)
-    tt_output = to_ttnn(torch_output, device=device)
+    tt_weight = to_ttnn(torch_weight, device=device, dtype=ttnn_dtype)
+    tt_output = to_ttnn(torch_output, device=device, dtype=ttnn_dtype)
```

Comment:
I think it's better to use `ttnn.from_torch`

Reply:
Same as the cmt above.